### DISCUSSION TITLE
The import-module now takes more time

### DISCUSSION THREAD

**Discussion by Daniel891** [dataplat/dbatools#8544] [General] [answered]
I've been updating the module to the 1.1.129 version because it fixes some features needed. But since then, any execution that use the module takes like 3 more times to complete. And I found out that it's the import-module what now takes more time.

In addition if I import just the functions I need (4 functions), it takes the same time. Btw I also tried the last version and still happening the same.

Do you know how could I solve this?   Could be fixed in some next version?

Thanks

**Comment by potatoqualitee** [comment]
I haven't noticed this issue @Daniel891  :/ What was the last version that was fast for you?

Yeah, I'm not really sure what the benefits of the specific command imports are. It's never sped things up for me either.

Execution policies have an impact on import time, perhaps yours changed? Set it to Bypass and see what happens? Also, I'd try Verbose to see where the slowdown happens.

**Reply by Daniel891** [reply]
The last version that was fast is the 1.1.70. 

I'm not saying that I use the specific command. I mean that anytime the module is used in a script, the module needs to be imported in the powershell session. And there is no difference if it's done manually or automatically or if I'm using a powershell profile. The script takes more time just because of the import process.

I still reaching out how could I solve it. Any help is welcome.

**Reply by potatoqualitee** [reply]
I don't recall any huge changes since then. Have you tried setting your execution policy to bypass? Execution policies are for safety, not security, so it's okay to set it to bypass so long as you pay attention to the scripts you are running.

Once you do, you can close PS then:

```
Import-Module dbatools
Measure-DbatoolsImport | Select Action, Duration
```

This is what mine looks like after setting my executionpolicy to bypass on Windows Server 2022. You can also run `$error` to see if you are running into hidden errors.

![image](https://user-images.githubusercontent.com/8278033/191478387-b330a1b1-c2ce-406d-9d22-ef1795501601.png)

**Reply by Daniel891** [reply]
The execution policy was set to bypass from the start.

I tried the Measure-DbatoolsImport but it's missing something because watching the clock the import really takes 30s.

![image](https://user-images.githubusercontent.com/67869983/191487872-c89aad93-ec67-4bce-abbf-cc016b753180.png)

**Comment by Daniel891** [comment, marked as answer]
Okey finally I found what is happening. 
For some reason the result is different depending how you install de module. I thought that the method was irrelevant and didn't try another one until now. At first I was downloading the zip from the web page and then just unzip on the path.
 But I tried with "install-module dbatools" and now the import-module just take 7s. 

I don't know if this is usual or why works like that but at least solve my problem.

Anyway thank you @potatoqualitee for your trying.

**Reply by potatoqualitee** [reply]
I should have considered that and will next time! Yes, we optimize it when it's in the gallery. Installs from git are more geared towards developers (sorta like Debug mode). So glad it's more reasonable!
